Work branch: Auth module

erastian opened the pull request:
- Changed .env CLIENT_URL var. Added http:// prefix before url for unifying use.
- In auth added forgot-password and restore-password methods
- For forgot-password and restore-password methods was added related methods in auth.service, users.service and users.repository
- Changed related interfaces
- Added template and method in mailer service for sending password restoration link
- In auth.controller fixed misspelled name of users.service
- Minor but useful fixes for mailer & auth module 
- Shifted code to try/catch block

erastian committed:
Work branch: Auth module

- Changed .env CLIENT_URL var. Added http:// prefix before url for unifying use.
- In auth added forgot-password and restore-password methods
- For forgot-password and restore-password methods was added related methods in auth.service, users.service and users.repository
- Added template and method in mailer service for sending password restoration link

mdziarkach commented on src/auth/auth.controller.ts:
```diff
@@ -86,16 +86,34 @@ export class AuthController extends BaseController implements IAuthController {
+		const { newPassword } = req.body;
+
+		try {
+			await this.authService.restorePassword(user, newPassword);
```
please provide only the necessary information for the service
```js
await this.authService.restorePassword(user.id, newPassword);
```

erastian replied:
Done

mdziarkach commented on src/mailer/mailer.service.ts:
```diff
@@ -60,4 +60,28 @@ export class MailerService implements IMailerService {
+				server_name: SERVER_NAME,
+				link: link,
+			},
+		};
```
move please code above inside of `try` section

erastian replied:
Done

erastian committed:
Work branch: Auth module

- mailer: put part of logic to try/catch block
- auth controller & service: forwarding from controller to service only user ID instead of forwarding whole user from request

mdziarkach commented on src/auth/auth.controller.ts:
```diff
@@ -86,16 +86,34 @@ export class AuthController extends BaseController implements IAuthController {
+			this.ok(res, 'Password was successfully changed');
+		} catch (e) {
+			return next(e);
+		}
```
```suggestion
try {
  const user = req.user;
  const { newPassword } = req.body;
  await this.authService.restorePassword(user.id, newPassword);

  this.ok(res, 'Password was successfully changed');
} catch (e) {
  return next(e);
}
```

erastian replied:
Done

erastian committed:
Work branch: Auth module

- Shifted code to try/catch block
